aws/docker.ts

```diff
@@ -296,30 +304,34 @@ interface CommandResult {
     stdout?: string;
 }
 
+let cliStreamID = -1;
```

Comment:
Why are these negative?

Also - due to node module loading of different versions, we can’t guarantee that we won’t load two separate copies of this module with separate counter, leading to overlapping stream IDs. If we need more guarantee that these are unique, we’ll likely need randomness or hash of some part of the operation identity.

Reply:
Also - perhaps a comment here about what the contract is for this.

Reply:
Bleagh.  Ok.  We have two approaches here.  Choose something random, or ask the host to actually give us back a stream object so that the host can guarantee uniqueness.

Reply:
Great point aobut loading different version.  I've switched to 'random' and have tried to comment heavily what's going on here.